engines/router/missionctl/log/resultlog/kafka_test.go

```diff
@@ -84,60 +85,74 @@ func TestNewKafkaLogger(t *testing.T) {
 	mockProducer.AssertCalled(t, "GetMetadata", &cfg.Topic, false, 1000)
 }
 
-func TestNewKafkaLogEntry(t *testing.T) {
+func TestNewJSONKafkaLogEntry(t *testing.T) {
 	// Create test Turing log entry
 	ctx, turingLogEntry := makeTestTuringResultLogEntry(t)
 
-	// Run newKafkaLogEntry and validate
-	logEntry, err := newKafkaLogEntry("test-app-name", turingLogEntry)
+	// Run newJSONKafkaLogEntry and validate
+	message, err := newJSONKafkaLogEntry("test-app-name", turingLogEntry)
 	assert.NoError(t, err)
 	// Get Turing request id
 	turingReqID, err := turingctx.GetRequestID(ctx)
 	tu.FailOnError(t, err)
-	// Set the Timestamp on logEntry, before comparison
-	logEntry.Timestamp = "test-timestamp"
 	// Compare logEntry data
-	assert.Equal(t, kafkaLogEntry{
-		RouterVersion: "test-app-name",
-		TuringReqID:   turingReqID,
-		Timestamp:     "test-timestamp",
-		Request: requestLogEntry{
-			Header: &http.Header{
-				"Req_id": []string{"test_req_id"},
+	assert.JSONEq(t,
+		fmt.Sprintf(`{
+			"router_version": "test-app-name",
+			"turing_req_id": "%s",
+			"event_timestamp": "2000-02-01T04:05:06.000000007Z",
+			"request": {
+				"header": {"Req_id": ["test_req_id"]},
+				"body": {"customer_id": "test_customer"}
 			},
-			Body: []byte(`{"customer_id": "test_customer"}`),
-		},
-		Experiment: &responseLogEntry{
-			Response: []byte(`{"key": "experiment_data"}`),
-			Error:    "",
-		},
-		Router: &responseLogEntry{
-			Response: []byte(`{"key": "router_data"}`),
-			Error:    "",
-		},
-		Enricher: &responseLogEntry{
-			Response: []byte(`{"key": "enricher_data"}`),
-			Error:    "",
-		},
-		Ensembler: nil,
-	}, *logEntry)
+			"experiment": {
+				"response": {"key": "experiment_data"}
+			},
+			"enricher": {
+				"response": {"key": "enricher_data"}
+			},
+			"router": {
+				"response": {"key": "router_data"}
```

Comment:
Do you think we should add a test case where error is not empty?

Reply:
Done